### PR TITLE
TMinus1Balance recalculations (with dry run flag enabled). BalanceT2 recalculations disabled.

### PR DESCRIPTION
**Pull request by ice-myles**
TMinus1Balance recalculations (with dry run flag enabled). BalanceT2 recalculations disabled.

### CODE REVIEW

File: miner/miner.go [hunk 1]
```diff
@@ -221,6 +261,14 @@ func (m *miner) mine(ctx context.Context, workerNumber int64) {
 			2. Fetching T0 & T-1 referrals of the fetched users.
 		******************************************************************************************************************************************************/
 
+		if balanceForTMinusBugfixEnabled || balanceForTMinusBugfixDryRunEnabled {
+			var err error
+			balances, err = m.dwhClient.GetBaseBalanceForTMinus1(ctx, userHistoryKeys, int64(len(userKeys)), 0)
```
Comment (ice-ares) [inline miner/miner.go]: remove `m.`

File: miner/recalculate_balance.go [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+	newUser.ID = usr.ID
+	newUser.UserID = usr.UserID
+	newUser.IDT0 = usr.IDT0
+	newUser.IDTMinus1 = usr.IDTMinus1
+	newUser.BalanceLastUpdatedAt = nil
+
+	return &newUser
+}
+
+func (m *miner) gatherHistory(ctx context.Context, users []*user, referralIDTMinus1Keys []string) (history map[int64][]*dwh.AdjustUserInfo, err error) {
```
Comment (ice-ares) [inline miner/recalculate_balance.go]: Pass dwhClient as arg

**Review by ice-ares — CHANGES_REQUESTED** [review]

File: miner/contract.go [hunk 3]
```diff
@@ -119,12 +132,17 @@ type (
 	}
 
 	recalculateReferral struct {
```
Comment (ice-ares) [inline miner/contract.go:134]: What are u doing with this?

Comment (ice-myles) [reply miner/contract.go:134]: It's t2 referral information to use for SUM(balanceTMinus1) for balanceT2 recalculations for each user in the batch. This data is taken from redis.

Comment (ice-ares) [reply miner/contract.go:134]: Yea but are u using this in storage.Save? Or in storage.Get/Bind?

Comment (ice-myles) [reply miner/contract.go:134]: `storage.Bind`

**Pull request merged**